File: server/src/main/java/com/koushik/movieflix/service/CommentService.java
package com.koushik.movieflix.service;

import java.util.List;

import com.koushik.movieflix.entities.CommentEntity;

public interface CommentService {
	public List<CommentEntity> showAll();

	public List<CommentEntity> showCommentsByMovie(String mid);

	public CommentEntity createComment(String userId, String movieId, CommentEntity entity);
}

File: client/src/app/services/comment.service.js
/**
 * Created by Koushik on 12/28/2016.
 */
(function () {
    angular.module('movieflix')
        .service('CommentService',CommentService);

    CommentService.$inject = ['$http','CONFIG'];
    function CommentService($http,CONFIG) {

        this.comment = function(userid,movieid,comment){
            var config = {
                url:CONFIG.API_HOST+'/comments/user/'+userid+'/title/'+movieid,
                method:'POST',
                data:comment
            }

            return $http(config)
                .then(function (resp) {
                    return resp.data;
                },function (err) {
                    return err;
                })
        }
    }
})();
